### PR TITLE
feat: DX12 support 

### PR DESCRIPTION
**Pull request by sindharta**
Supports both hardware and software encoding

### CODE REVIEW

File: Plugin~/WebRTCPluginTest/GraphicsDeviceTest.cpp [hunk 1]
```diff
@@ -43,8 +43,6 @@ TEST_P(GraphicsDeviceTest, CopyResourceV) {
     const auto height = 256;
     const auto src = m_device->CreateDefaultTextureV(width, height);
     const auto dst = m_device->CreateDefaultTextureV(width, height);
-    EXPECT_TRUE(m_device->CopyResourceV(dst, src));
-    EXPECT_FALSE(m_device->CopyResourceV(src, src));
```
Comment (kannan-xiao4) [inline Plugin~/WebRTCPluginTest/GraphicsDeviceTest.cpp]: Why delete these expect ?

File: Plugin~/WebRTCPlugin/GraphicsDevice/D3D12/D3D12Texture2D.cpp [hunk 2]
```diff
@@ -9,8 +11,44 @@ D3D12Texture2D::D3D12Texture2D(uint32_t w, uint32_t h, ID3D12Resource* nativeTex
...
+    desc.Format = DXGI_FORMAT_UNKNOWN;
+    desc.SampleDesc.Count = 1;
+    desc.SampleDesc.Quality = 0;
+    desc.Layout = D3D12_TEXTURE_LAYOUT_ROW_MAJOR;
+    desc.Flags = D3D12_RESOURCE_FLAG_NONE;
+
+    const HRESULT hr = device->CreateCommittedResource(&D3D12_READBACK_HEAP_PROPS, D3D12_HEAP_FLAG_NONE,
+        &desc, D3D12_RESOURCE_STATE_COPY_DEST, nullptr, IID_PPV_ARGS(&m_readbackResource)
+    );
+
```
Comment (karasusan) [inline Plugin~/WebRTCPlugin/GraphicsDevice/D3D12/D3D12Texture2D.cpp]: It should check the status of `HRESULT` and return the success/failure to the caller.

File: Plugin~/WebRTCPlugin/Codec/EncoderFactory.cpp [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
 #include "NvCodec/NvEncoderCuda.h"
 
+#include "NvCodec/NvEncoderD3D12.h"
```
[the remark this answers is not in the document]
Comment (sindharta) [reply Plugin~/WebRTCPlugin/Codec/EncoderFactory.cpp]: Will fix

File: Plugin~/WebRTCPlugin/GraphicsDevice/D3D12/D3D12Texture2D.cpp (hunk 2 above)
Comment (sindharta) [reply Plugin~/WebRTCPlugin/GraphicsDevice/D3D12/D3D12Texture2D.cpp]: will fix

**Comment by karasusan** [comment]
LGTM, but should remove conflicting file.

**Pull request merged**